lxd/storage_zfs.go

```diff
@@ -63,6 +63,13 @@ func (s *storageZfs) StorageCoreInit() error {
 	}
 	s.sTypeName = typeName
 
+	if !shared.RunningInUserNS() {
```

Comment:
Why is that needed? Seems like an extra, somewhat lengthy check for no real good reason.

If someone attempts to use zfs inside a container, they most likely have the module loaded on the host which would make this code succeed and if it's not loaded on the host, things will fail anyway so may as well be the module loading failure be the first :)